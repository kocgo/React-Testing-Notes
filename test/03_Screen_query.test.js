import * as React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom/extend-expect";
import Counter from "../src/components/counter";

test("counter increments and decrements when the buttons are clicked", () => {
  const { container } = render(<Counter />);
  // Screen queries (Accesibility properties)
  // (`getByRole` can work for the button too)
  const incrementButton = screen.getByText("Increment");
  const decrementButton = screen.getByRole("button", { name: /decrement/i });
  const message = container.firstChild.querySelector("div");

  expect(message).toHaveTextContent("Current count: 0");
  fireEvent.click(incrementButton);
  expect(message).toHaveTextContent("Current count: 1");
  // Same thing with userEvent
  userEvent.click(decrementButton);
  expect(message).toHaveTextContent("Current count: 0");
});
