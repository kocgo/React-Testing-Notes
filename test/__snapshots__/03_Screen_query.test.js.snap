// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter increments and decrements when the buttons are clicked 1`] = `
<div>
  Current count: 
  0
</div>
`;
